src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -105,12 +126,78 @@ _IRQL_requires_max_(DISPATCH_LEVEL)
 _Function_class_(CXPLAT_DATAPATH_RECEIVE_CALLBACK)
 void
 UdpRecvCallback( 
-    _In_ CXPLAT_SOCKET* /* Binding */,
-    _In_ void* /* Context */,
+    _In_ CXPLAT_SOCKET* Binding,
+    _In_ void* Context,
     _In_ CXPLAT_RECV_DATA* RecvBufferChain
     )
 {
-    CxPlatRecvDataReturn(RecvBufferChain);
+    CXPLAT_RECV_DATA* Datagram;
+    CXPLAT_RECV_DATA* CopyRecvBufferChain = RecvBufferChain;
```

Comment:
This doesn't copy the whole chain, just the first item. When you do through the `while` loop below, you set each `Next` pointer to `NULL`, effectively breaking this chain's copy. I _think_ you just need to not set `Next` to `NULL` and then this will work. But if you do that, then you also don't need 3 variables (`RecvBufferChain`, `CopyRecvBufferChain`, `Datagram`). Remove the copy one, and just use `Datagram` to walk the list, then free it all when you're done.